src/cli/helpers.ts

```diff
@@ -119,8 +120,37 @@ export async function emitMetaMessage(
 ): Promise<void> {
   // eslint-disable-next-line @typescript-eslint/no-var-requires
   const { version } = require('../../package.json')
+  const ciEnvironment = detectCiEnvironment(env)
+  const meta: messages.Meta = {
+    protocolVersion: messages.version,
+    implementation: {
+      version,
+      name: 'cucumber-js',
+    },
+    cpu: {
+      name: os.arch(),
+    },
+    os: {
+      name: os.platform(),
+      version: os.release(),
+    },
+    runtime: {
+      name: 'node.js',
+      version: process.versions.node,
+    },
+    ci: ciEnvironment
+      ? {
+          ...ciEnvironment,
+          git: {
+            ...ciEnvironment.git,
+            remote: ciEnvironment.git.remote ?? '',
```

Comment:
@aslakhellesoy these fields are mandatory in `messages.Meta` but optional in `CiEnvironment`. Could we also make them optional in messages?

Reply:
@davidjgoss I think we should make `Git.remote` and `Git.revision` mandatory in `ci-environment` (they already are in `messages`). These fields are the bare minimum for using the data and if either of them are absent we shouldn't create a `Git` object at all. Both `branch` and `tag` can be undefined.

Reply:
See https://github.com/cucumber/ci-environment/issues/36

Reply:
Updated with the new library - much simpler now!